src/routes/view-space.js

```diff
@@ -82,6 +82,42 @@ const SurveyJumbo = (props) => (
     </Jumbotron>
 );
 
+/**
+ * Improvements:
+ *  - animated heart icon for the favorites button (see: https://codepen.io/mapk/pen/ZOQqaQ)
+ *  - better background color
+ *  - include a toast (maybe "Added to Favorites") once clicked
+ *  - disable button as soon as it's clicked instead of having to refresh the page
+ *  - favorites button on the main nav bar should automatically requery for the
+ *    local storage data instead of having to do a manual refresh
+ */
+const FavoritesButton = (props) => {
+    let favoriteSpacesJson = JSON.parse(localStorage.getItem('favoriteSpaces'));
+
+    console.log(typeof favoriteSpacesJson.data.map(space => space.title).includes(props.space.title))
+
+    /* Add the currently viewed space to the favorites list */
+    function handleClick() {
+        favoriteSpacesJson.data.push(props.space);
+        console.log(favoriteSpacesJson);
+        localStorage.setItem('favoriteSpaces', JSON.stringify(favoriteSpacesJson));
+    }
+
+    /* Disable button if a space has already been added to the favorites list */
+    function shouldDisable() {
+        return favoriteSpacesJson.data.map(space => space.title).includes(props.space.title);
+    }
+
+    return (<Button
+      style={{ position: "absolute", top: "100px", right: "20px" }}
+      variant="secondary"
+      onClick={handleClick}
+      disabled={shouldDisable()}
+      >
+      Add to Favorites
+    </Button>);
+};
+
```

Comment:
could this possibly be its own component/file?

Reply:
It could but I feel like it's not going to used in other components so I thought it would be better to keep it local to the only component that is currently using it. How do you feel about that? I can move it into a separate file if needed.

Reply:
i feel like you could make it into a component and have it be shared with `favorites.js` so they would appear consistent.

Reply:
ok sounds good, will do